### PR TITLE
Moving events to v2

### PR DESCRIPTION
**Pull request by iovadia**
Signed-off-by: iovadia <iovadia@redhat.com>

/cc @osherdp @eliorerz @lalon4 @nshidlin

### CODE REVIEW

File: discovery-infra/test_infra/helper_classes/cluster.py [hunk 1]
```diff
@@ -950,8 +950,8 @@ def _get_namespace_index(cls, libvirt_network_if):
         matcher = re.match(r"^tt(\d+)$", libvirt_network_if)
         return int(matcher.groups()[0]) if matcher is not None else 0
 
-    def _find_event(self, event_to_find, reference_time, params_list, host_id):
-        events_list = self.get_events(host_id=host_id)
+    def _find_event(self, event_to_find, reference_time, params_list, host_id, infra_env_id):
```
Comment (eliorerz) [inline discovery-infra/test_infra/helper_classes/cluster.py:953]: Add type hint annotations and return type annotations

File: discovery-infra/test_infra/helper_classes/cluster.py [hunk 2]
```diff
@@ -963,19 +963,19 @@ def _find_event(self, event_to_find, reference_time, params_list, host_id):
         else:
             return False
 
-    def wait_for_event(self, event_to_find, reference_time, params_list=None, host_id="", timeout=10):
+    def wait_for_event(self, event_to_find, reference_time, params_list=None, host_id="",infra_env_id="", timeout=10):
```
Comment (eliorerz) [inline discovery-infra/test_infra/helper_classes/cluster.py:966]: Add type hint annotations

File: discovery-infra/test_infra/helper_classes/cluster.py [hunk 3]
```diff
@@ -926,8 +926,8 @@ def reset_cluster_and_wait_for_ready(cls, cluster):
         cluster.wait_until_hosts_are_discovered()
         cluster.wait_for_ready_to_install()
 
-    def get_events(self, host_id=""):
-        return self.api_client.get_events(cluster_id=self.id, host_id=host_id)
+    def get_events(self, host_id="", infra_env_id=""):
```
Comment (eliorerz) [inline discovery-infra/test_infra/helper_classes/cluster.py:929]: Add type hint annotations and return type annotations

**Comment by eliorerz** [comment]
I'm not sure events should stay under `Cluster` ass it was before.
On v2 `events` endpoint moved to independent endpoint and no longer is under `cluster`.
I think we should create new object to handle the events.
@nmagnezi @tsorya @nshidlin WDYT?

**Comment by tsorya** [comment]
> I'm not sure events should stay under `Cluster` as it was before. On v2 `events` endpoint moved to independent endpoint and it's no longer under `cluster`. I think we should create new object to handle the events.
> 
> e.g.
> 
> ```python
> class EventsHandler:
>     def __init__(self, api_client: InventoryClient, cluster_id: str, infra_env_id: str):
>         self.api_client = api_client
>         self.infra_env_id = infra_env_id
>         self.cluster_id = cluster_id
> 
>     def _find_event(self, event_to_find, reference_time, params_list, host_id):
>         ...
>     def get_events(self, host_id=""):
>         ...
> 
>     def wait_for_event(self, event_to_find, reference_time, params_list=None, host_id="", timeout=10):
>         ...
> ```
> 
> @nmagnezi @tsorya @nshidlin WDYT?

I don't think we need events out of the cluster scope so i don't see any reason to move 2 small api out of it right now

**Review by nmagnezi — CHANGES_REQUESTED** [review]
> I'm not sure events should stay under `Cluster` as it was before. On v2 `events` endpoint moved to independent endpoint and it's no longer under `cluster`. I think we should create new object to handle the events.
> 
> e.g.
> 
> ```python
> class EventsHandler:
>     def __init__(self, api_client: InventoryClient, cluster_id: str, infra_env_id: str):
>         self.api_client = api_client
>         self.infra_env_id = infra_env_id
>         self.cluster_id = cluster_id
> 
>     def _find_event(self, event_to_find, reference_time, params_list, host_id):
>         ...
>     def get_events(self, host_id=""):
>         ...
> 
>     def wait_for_event(self, event_to_find, reference_time, params_list=None, host_id="", timeout=10):
>         ...
> ```
> 
> @nmagnezi @tsorya @nshidlin WDYT?

IMO, you make a good point. In V1 all events had `cluster-id`, but now this is no longer the case. For example, host events that are linked to `infra_env`, but might not be bounded to a cluster at this point.

https://github.com/openshift/assisted-service/blob/cc39df66bb6431e7843d3be828fdc9e4a731939c/docs/events.yaml#L35-L41

**Comment by eliorerz** [comment]
> In that case lets move them with another jira ticket. Right now there is no use of events without cluster id in the code. Lets finish this task and move them to new object when we will need it

OK,  don't think it requires much of a change but I opened a ticket for doing it https://issues.redhat.com/browse/MGMT-7977

**Review by eliorerz — APPROVED** [review]

**Pull request merged**